ports/views.py

```diff
@@ -205,92 +259,89 @@ def maintainer_detail_email(request, name, domain):
-        search_in = request.POST['categories__name']
-
-        filtered_ports = PortFilterByMultiple(request.POST, queryset=Port.objects.all()).qs[:50]
-        return render(request, 'ports/filtered_table.html', {
-            'ports': filtered_ports,
-            'query': query,
-            'search_in': search_in,
-            'content': "Category"
-            })
-    else:
-        return HttpResponse("Method not allowed")
+    query = request.GET.get('name')
+    search_in = request.GET.get('categories__name')
+
+    filtered_ports = PortFilterByMultiple(request.GET, queryset=Port.objects.all()).qs[:50]
+    return render(request, 'ports/ajax-filters/filtered_table.html', {
+        'ports': filtered_ports,
+        'query': query,
+        'search_in': search_in,
+        'content': "Category"
+    })
 
 
 # Respond to ajax calls for searching within a maintainer
 def search_ports_in_maintainer(request):
-    if request.method == 'POST':
-        query = request.POST['name']
-        search_in = request.POST['maintainers__name']
-
-        filtered_ports = PortFilterByMultiple(request.POST, queryset=Port.objects.all()).qs[:50]
-        return render(request, 'ports/filtered_table.html', {
-            'ports': filtered_ports,
-            'query': query,
-            'search_in': search_in,
-            'content': "Maintainer"
-        })
-    else:
-        return HttpResponse("Method not allowed")
+    query = request.GET.get('name', '')
+    search_in = request.GET.get('maintainers__name', '')
+
+    filtered_ports = PortFilterByMultiple(request.GET, queryset=Port.objects.all()).qs[:50]
+    return render(request, 'ports/ajax-filters/filtered_table.html', {
+        'ports': filtered_ports,
+        'query': query,
+        'search_in': search_in,
+        'content': "Maintainer"
+    })
+
+
+def search_ports_in_variant(request):
+    query = request.GET.get('name', '')
+    search_in = request.GET.get('variant', '')
+
+    filtered_ports = Variant.objects.filter(variant=search_in, port__name__icontains=query)
+    return render(request, 'ports/ajax-filters/filtered_table.html', {
+        'ports': filtered_ports,
+        'query': query,
+        'search_in': search_in,
+        'content': "Variant"
+    })
 
 
-# Accept submissions from mpstats and update the users table
 @csrf_exempt
 def stats_submit(request):
     if request.method == "POST":
         try:
             received_json = json.loads(request.POST.get('submission[data]'))
-
-            user = User()
-            user.uuid = received_json['id']
-            user.osx_version = received_json['os']['osx_version']
-            user.macports_version = received_json['os']['macports_version']
-            user.os_arch = received_json['os']['os_arch']
-            user.xcode_version = received_json['os']['xcode_version']
-            user.full_json = received_json
-            user.save()
+            submission_id = Submission.populate(received_json)
```

Comment:
You can get the `submission_timestamp` here using `datetime.datetime.utcnow()` and store it in the submissions table.